src/signals-unix.c

```diff
@@ -600,6 +600,7 @@ static void *signal_listener(void *arg)
         if (sigwait(&sset, &sig)) {
             sig = SIGABRT; // this branch can't occur, unless we had stack memory corruption of sset
         }
+#ifdef _OS_DARWIN_
```

Comment:
Hmmm, should `sigwait` even set `errno`?

The Linux and POSIX man page sayas it should return 0 or the error number and did not mention anything about `errno`.

I believe in general unless the function is explicitly specified to set `errno` the value in it shouldn't really mean anything so is this just polking into some darwin libc internals?

Reply:
Right, but it turns out that Linux is sometimes setting errno (to EINTR). I guess that's allowed, though a bit unexpected. On Darwin, this doesn't work very well either, but usually we use kevent (this is just in the fallback code if that failed to initialize).